pkg/auth/audit/filter.go

```diff
@@ -65,14 +73,29 @@ func (h auditHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	auditLog, err := newAuditLog(h.auditWriter, req, h.sanitizingRegex)
 	if err != nil {
-		util.ReturnHTTPError(rw, req, 500, err.Error())
+		util.ReturnHTTPError(rw, req, http.StatusInternalServerError, err.Error())
 		return
 	}
 
 	wr := &wrapWriter{ResponseWriter: rw, auditWriter: h.auditWriter, statusCode: http.StatusOK}
 	h.next.ServeHTTP(wr, req)
 
-	auditLog.write(user, req.Header, wr.Header(), wr.statusCode, wr.buf.Bytes())
+	err = auditLog.write(user, req.Header, wr.Header(), wr.statusCode, wr.buf.Bytes())
+	if err == nil {
+		return
+	}
+
+	// Locking after next is called to avoid performance hits on the request.
+	h.errLock.Lock()
+	defer h.errLock.Unlock()
+
+	// Only log duplicate error messages at most every errorDebounceTime.
+	// This is to prevent the rancher logs from being flooded with error messages
+	// when the log path is invalid or any other error that will always cause a write to fail.
+	if lastSeen, ok := h.errMap[err.Error()]; !ok || time.Since(lastSeen) > errorDebounceTime {
```

Comment:
I'm not a huge fan of this logic. IMO, it has the following disadvantages:
- You are maintaining an ever growing map of errors. So if we have a lot of slightly different errors, your map will keep getting bigger.
- The map is never pruned, even when the lastSeen time exceeds the debounce time. So this map will sit in memory and continue to grow over time.

From an overall perspective, I also don't think the issue here needs to be solved. It may be annoying when errors flood the logs, but it ends up being very clear about the fact that there is an error and that you need to fix it. In my opinion, the slight annoyance caused by this doesn't necessitate the overhead of us creating/maintaining a solution. 

But that's just me. I don't think that this is vital enough to hold up your PR (though I do think we will want to prune that map eventually - I could see that causing a non-0 memory uptick), but I did want to bring it up.